system/kube-monitoring/charts/prometheus-frontend/templates/_kubernetes.alerts.tpl

```diff
@@ -321,3 +317,29 @@ ALERT KubernetesTooManyOpenFiles
     summary = "Too many open file descriptors",
     description = "{{`{{$labels.job}}`}} on {{`{{$labels.instance}}`}} is using {{`{{$value}}`}}% of the available file/socket descriptors.",
   }
+
+ALERT HighNumberOfGoRoutines
+  IF irate(go_goroutines{job="kube-system/kubelet"}[5m]) > 20
```

Comment:
Generally it's recommened to use `rate` for alerting. 

> irate should only be used when graphing volatile, fast-moving counters. Use rate for alerts and slow-moving counters, as brief changes in the rate can reset the FOR clause and graphs consisting entirely of rare spikes are hard to read.

Looks like a good heuristic. Though I could image quite a few false positives. What would you think about something like:

```
go_goroutines{job="kube-system/kubelet"} > avg_over_time(go_goroutines{job="kube-system/kubelet"}[12h] offset 12h) * 2
```

This would alert if there is suddenly twice as many goroutines than over the average of the last 12 hours ago.